lib/PEX.ts

```diff
@@ -507,4 +329,20 @@ export class PEJS {
     }
     return { version: version };
   }
+
+  private determineAndCastToInternalPresentationDefinition(
+    presentationDefinition: PresentationDefinitionV1 | PresentationDefinitionV2
+  ): InternalPresentationDefinition {
+    const versionResult: { version?: PEVersion; error?: string } =
+      this.definitionVersionDiscovery(presentationDefinition);
```

Comment:
For the definitonVersionDiscovery: some other changes are present between v1 and v2. We probably might consider checking against the respective schemas instead of doing manual checks. Check against v2 schema with ajv. If that works use v2. If not check against v1. If that works use v1. If not throw an error


See for instance the addition of format schemas they have done, so we can validate between jwt and ldp based formats:
https://identity.foundation/presentation-exchange/#format-declaration

These are the differences in schemas for definitions:

v1 def: https://identity.foundation/presentation-exchange/spec/v1.0.0/#json-schema-2 vs v2 defhttps://identity.foundation/presentation-exchange/#presentation-definition-2

Reply:
tried to use `Ajv` for checking object vs schema, unfortunately it's not possible, noticed that Ajv doesn't complain when we have extra properties in the object.
Now I'm passing the object to both v1 and v2 validators to see which one doesn't complain.

Reply:
Have you looked at the additionalProperties flag in an object? That controls whether additional properties are allowed or not: http://json-schema.org/understanding-json-schema/reference/object.html#id5

Reply:
yes, as stated in the link that you've sent:

> By default, providing additional properties is valid

but my understanding is to make it invalid, we have to use the flag: `"additionalProperties": false` but even with that, we're passing the schema check.

Reply:
@sksadjad Do you have a standalone testcase to show that behaviour. So simply the schema together with an input and ajv. Because that would be rather odd IMO

Reply:
yes, I wrote one simple one the other day with @hrehman-sphereon but didn't push it

Reply:
Please do push it, so I can do a validation

Reply:
pushed. the `$schema` property of the schema is 'http://json-schema.org/draft-07/schema#'. Ajv doesn't validate this schema, and returns true for the data in test (test/validation/core/ajvSchema.spec.ts)

Reply:
After internal testing we resolved this to use ajv